sync/block_listener.go

```diff
@@ -92,51 +91,38 @@ func (bl *BlockListener) ListenToGossipBlocks() {
 	}
 }
 
-func (bl *BlockListener) handleBlock(blk *types.Block) bool {
+func (bl *BlockListener) HandleNewBlock(blk *types.Block) bool {
 
 	bl.Log.With().Info("got new block", log.Uint64("id", uint64(blk.Id)), log.Int("txs", len(blk.TxIds)), log.Int("atxs", len(blk.AtxIds)))
-	eligible, err := bl.BlockEligible(&blk.BlockHeader)
+	//check if known
+	if _, err := bl.GetBlock(blk.Id); err == nil {
+		bl.Info("we already know this block %v ", blk.ID())
+		return true
+	}
+
+	if err := bl.ValidateBlock(blk); err != nil {
```

Comment:
shouldn't we also check the blocks creator eligibility same as previously?

Reply:
you can discuss the eligibility flow with bark its going to be a block box and handled outside the sync